rust/sbp/src/sbp_tools.rs

```diff
@@ -1,13 +1,52 @@
+use std::{collections::HashMap, convert::TryFrom, fmt::Debug};
+
 #[cfg(feature = "swiftnav-rs")]
 use swiftnav_rs::time::GpsTime;
 
 #[cfg(feature = "swiftnav-rs")]
-use crate::{
-    messages::SBPMessage,
-    time::{GpsTimeError, MessageTime, RoverTime},
-};
+use crate::time::{GpsTimeError, MessageTime, RoverTime};
+
+use crate::messages::{SBPMessage, SBP};
+
+pub struct Dispatcher {
```

Comment:
I'm not sure if this actually belongs in libsbp, just want to show what I'd like the end result to be

Reply:
I'm not sure if the libsbp repo is the right place, but I definitely could see this being a separate crate (`sbp-dispatcher` maybe?).